app/controllers/application_controller.rb

```diff
@@ -217,26 +206,21 @@ def fix_broken_personal_key_url
   end
 
   def after_sign_in_path_for(_user)
-    accept_rules_of_use_url ||
-      user_suspended_url ||
-      service_provider_mfa_setup_url ||
-      add_piv_cac_setup_url ||
-      fix_broken_personal_key_url ||
-      user_session.delete(:stored_location) ||
-      sp_session_request_url_with_updated_params ||
-      signed_in_url
-  end
-
-  def signed_in_url
-    return user_two_factor_authentication_url unless user_fully_authenticated?
+    return rules_of_use_path if !current_user.accepted_rules_of_use_still_valid?
+    return user_please_call_url if current_user.suspended?
+    return authentication_methods_setup_url if user_needs_sp_auth_method_setup?
+    return login_add_piv_cac_prompt_url if session[:needs_to_setup_piv_cac_after_sign_in].present?
+    return fix_broken_personal_key_url if current_user.broken_personal_key?
+    return user_session.delete(:stored_location) if user_session.key?(:stored_location)
     return reactivate_account_url if user_needs_to_reactivate_account?
```

Comment:
After chatting with @mitchellhenke , this should probably be in `signed_in_url` after all, the thinking being that a user shouldn't be prompted for reactivation if they're not authenticating with an IdV request. And if they were authenticating with an IdV request, the existing logic for triggering IdV on the basis of lack of active profile would already happen.

The fact that tests are passing is alarming, though, if that was an expected behavior.

Reply:
After investigating further, I noticed that the `user_needs_to_reactivate_account?` check here is already accounting for the behavior described above where a user should only go through the reactivation flow if they're signing in to a partner requesting IdV:

https://github.com/18F/identity-idp/blob/50e6b15bf58665b34634f4b0da6577163805f3a9/app/controllers/application_controller.rb#L252-L256

So moving this into `after_signed_in_path_for` should not regress to start prompting IAL1 authentications with the profile reactivation. I pushed a temporary commit c2d627f42 to double-check that we have test coverage for this, though I've already put together some feature spec enhancements in case we don't.

Reply:
There's decent controller coverage here, though somewhat indirect, and only testing an `#update` path 🤔 

https://github.com/18F/identity-idp/blob/50e6b15bf58665b34634f4b0da6577163805f3a9/spec/controllers/users/personal_keys_controller_spec.rb#L74-L86

I think I'll go ahead and push my feature spec revisions.

Reply:
> I think I'll go ahead and push my feature spec revisions.

See [ef3f613](https://github.com/18F/identity-idp/pull/9159/commits/ef3f6135bb2b48c333bf9be6ee8534c4b67d1389)